common/browser_context_test.go

```diff
@@ -50,3 +51,240 @@ func TestNewBrowserContext(t *testing.T) {
+			},
+			cookies: []*api.Cookie{
+				{
+					Domain: "foo.com",
+				},
+				{
+					Domain: "bar.com",
+				},
+				{
+					Domain: "baz.com",
+				},
+			},
+			wantCookies: []*api.Cookie{
+				{
+					Domain: "foo.com",
+				},
+				{
+					Domain: "baz.com",
+				},
+			},
+		},
+		"filter_by_subdomain": {
+			filterByURLs: []string{
+				"https://sub.foo.com",
+			},
+			cookies: []*api.Cookie{
+				{
+					Domain: ".foo.com",
+				},
+			},
+			wantCookies: []*api.Cookie{
+				{
+					Domain: ".foo.com",
+				},
+			},
+		},
+		"filter_dot_prefixed_domains": {
+			filterByURLs: []string{
+				"https://foo.com",
+			},
+			cookies: []*api.Cookie{
+				{
+					Domain: ".foo.com",
+				},
+			},
+			wantCookies: []*api.Cookie{
+				{
+					Domain: ".foo.com",
+				},
+			},
+		},
+		"filter_by_secure_cookies": {
+			filterByURLs: []string{
+				"https://foo.com",
+			},
+			cookies: []*api.Cookie{
+				{
+					Domain: "foo.com",
+					Secure: true,
+				},
+			},
+			wantCookies: []*api.Cookie{
+				{
+					Domain: "foo.com",
+					Secure: true,
+				},
+			},
+		},
+		"filter_by_http_only_cookies": {
+			filterByURLs: []string{
+				"https://foo.com",
+			},
+			cookies: []*api.Cookie{
+				{
+					Domain:   "foo.com",
+					HTTPOnly: true,
+				},
+			},
+			wantCookies: []*api.Cookie{
+				{
+					Domain:   "foo.com",
+					HTTPOnly: true,
+				},
+			},
+		},
+		"filter_by_path": {
+			filterByURLs: []string{
+				"https://foo.com/bar",
+			},
+			cookies: []*api.Cookie{
+				{
+					Domain: "foo.com",
+					Path:   "/bar",
+				},
+				{
+					Domain: "foo.com",
+					Path:   "/baz",
+				},
+			},
+			wantCookies: []*api.Cookie{
+				{
+					Domain: "foo.com",
+					Path:   "/bar",
+				},
+			},
+		},
+		"disallow_secure_cookie_on_http": {
+			filterByURLs: []string{
+				"http://foo.com",
+			},
+			cookies: []*api.Cookie{
+				{
+					Domain: "foo.com",
+					Secure: true,
+				},
+			},
+			wantCookies: nil,
+		},
+		"invalid_filter": {
+			filterByURLs: []string{
+				"HELLO WORLD!",
+			},
+			cookies: []*api.Cookie{
+				{
+					Domain: "foo.com",
+				},
+			},
+			wantCookies: nil,
+			wantErr:     true,
+		},
+		"invalid_filter_empty": {
+			filterByURLs: []string{
+				"",
+			},
+			cookies: []*api.Cookie{
+				{
+					Domain: "foo.com",
+				},
+			},
+			wantCookies: nil,
+			wantErr:     true,
+		},
+		"invalid_filter_multi": {
+			filterByURLs: []string{
+				"https://foo.com", "", "HELLO WORLD",
+			},
+			cookies: []*api.Cookie{
+				{
+					Domain: "foo.com",
+				},
+			},
+			wantCookies: nil,
+			wantErr:     true,
+		},
+	}
+	for name, tt := range tests {
+		t.Run(name, func(t *testing.T) {
+			cookies, err := filterCookies(
+				tt.cookies,
+				tt.filterByURLs...,
+			)
+			if tt.wantErr {
+				assert.Nilf(t, cookies, "want no cookies after an error, but got %#v", cookies)
+				require.Errorf(t, err, "want an error, but got none")
+				return
+			}
+			require.NoError(t, err)
+
+			assert.Lenf(t,
```

Comment:
Do we need this length assertion? I would've thought the equal assertion below will deal with that and we will get a nice output of both the `want` and `got` to make debugging simpler.